Sources/PersistentCollections/Node/_Stack.swift

```diff
@@ -0,0 +1,115 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// A fixed-size array of just enough size to hold an ancestor path in a
+/// `PersistentDictionary`.
+@usableFromInline
+@frozen
+internal struct _Stack<Element> {
+#if arch(x86_64) || arch(arm64)
+  @inlinable
+  @inline(__always)
+  internal static var capacity: Int { 13 }
+
+  @usableFromInline
+  internal var _contents: (
+    Element, Element, Element, Element,
+    Element, Element, Element, Element,
+    Element, Element, Element, Element,
+    Element
+  )
+#else
+  @inlinable
+  @inline(__always)
+  internal static var capacity: Int { 7 }
+
+  // xxxxx xxxxx xxxxx xxxxx xxxxx xxxxx xx
+  @usableFromInline
+  internal var _contents: (
+    Element, Element, Element, Element,
+    Element, Element, Element
+  )
+#endif
```

Comment:
Is `hashValue` in Swift on 32-bit systems 32 bits as well?

Reply:
Yep! `Hasher` returns exactly `Int`-sized globs of bits.

Reply:
It's a shame we have to hardwire these lengths here, by the way, and also that we have to manually list architectures instead of just [directly asking the compiler about the size of `Int`](https://forums.swift.org/t/pitch-pointer-bit-width-compile-time-conditional/59572)...